tests/Entity/BaseEntityManagerTest.php

```diff
@@ -67,4 +68,19 @@ public function testGetEntityManager()
 
         $manager->em;
     }
+
+    public function testGetRepository()
+    {
+        $entityRepository = $this->createMock(EntityRepository::class);
+        $objectManager = $this->createMock(ObjectManager::class);
+        $objectManager->expects($this->once())->method('getRepository')->with('classname')->willReturn($entityRepository);
+
+        $registry = $this->createMock(ManagerRegistry::class);
+        $registry->expects($this->once())->method('getManagerForClass')->willReturn($objectManager);
+
+        $manager = new EntityManager('classname', $registry);
+        $method = new \ReflectionMethod($manager, 'getRepository');
+        $method->setAccessible(true);
+        $method->invokeArgs($manager, []);
```

Comment:
I would expect a test with assertInstanceOf here… Also, you could avoid reflection by adding a public method to `EntityManager` that would call `getRepository`